atef/procedure.py

```diff
@@ -81,8 +83,43 @@ def run(self) -> Result:
                 reason=str(ex)
             )
 
-        self.result = result
-        return result
+        # stash step result
+        self.step_result = result
+        # return the overall result, including verification
+        return self.result
+
+    @property
+    def result(self) -> Result:
```

Comment:
If this is something we want serialized (i.e., when saving procedure results/data), it might not belong in a property.

Reply:
I think I stash it under `_result` for that reason actually.  That should get updated every time the property gets called

Reply:
Could `_result` remain `result`, and then this be something like `overall_result` property or perhaps `get_display_result()` or `get_combined_result()` or something?

Reply:
That's right, making `result` a property backed by calculations was done to keep the interface between `Configuration` and `ProcedureStep` similar.  Both have a `result` that the `RunCheck` widget looks for

I was worried about future consumers grabbing an "overall result" that isn't updated with the most recent verify and step results.  I'd be satisfied with renaming `_result` -> `combined_result` and leaving property as is, but does that also jive with you?

Reply:
Renamed `_result` -> `combined_result` in [19287a7](https://github.com/pcdshub/atef/pull/150/commits/19287a72a182cc77520f34f2d58a4b8f49f1e901)